File: Algorithms/findMinimumInRotatedSortedArray.py
class Solution:
    # @param num, a list of integer
    # @return an integer
    def findMin(self, num):
        return self.findMin_interval(num, 0, len(num) - 1)
    def findMin_interval(self, num, start, end):
        if start == end:
            return num[start]
        if num[start] < num[end]:
            return num[start]
        mid = start + (end - start) / 2
        if num[mid] >= num[start]:
            return self.findMin_interval(num, mid+1, end)
        return self.findMin_interval(num, start, mid)

if __name__ == '__main__':
    s = Solution()
    print s.findMin([1])

File: Algorithms/findMinimumInRotatedSortedArray.cpp
#include "leetcode.h"
class Solution {
public:
    int findMin(vector<int> &num) {
        int start = 0, end = num.size()-1;
        while (start < end && num[start] > num[end]) {
            int mid = start + (end - start)/2;
            if (num[mid] > num[end]) {
                start = mid + 1;
            } else {
                end = mid;
            }
        }
        return num[start];
    }
    /*
    public:
    int findMin(vector<int> &num) {
        return findMin(num, 0, num.size()-1);
    }
    private:
    int findMin(vector<int> &num, int start, int end) {
        if (start == end)
            return num[start];
        if (num[start] < num[end])
            return num[start];
        int mid = start + (end - start) / 2;
        if (num[mid] >= num[start])
            return findMin(num, mid+1, end);
        return findMin(num, start, mid);
    }
    */
};

int main() {
    Solution s;
    //int nums[] = {4, 5, 6, 7, 1, 2};
    int nums[] = {1, 2};
    vector<int> num(nums, nums + sizeof(nums)/sizeof(nums[0]));
    cout << s.findMin(num) << endl;
    return 0;
}
